src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -3,17 +3,52 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+
+    private static final int MAX_STORAGE_NUMBER = 10;
+    private K[] keyArray;
+
+    private V[] valueArray;
+
+    private int size;
+
+    public StorageImpl() {
+        keyArray = (K[]) new Object[MAX_STORAGE_NUMBER];
+        valueArray = (V[]) new Object[MAX_STORAGE_NUMBER];
+        size = 0;
+    }
+
     @Override
     public void put(K key, V value) {
+        if (size == MAX_STORAGE_NUMBER) {
+            return;
+        }
+        int keyIndex = findIndex(key);
+        if (keyIndex >= 0) {
+            valueArray[keyIndex] = value;
+        } else {
+            keyArray[size] = key;
+            valueArray[size] = value;
+            size++;
+        }
+    }
+
+    private int findIndex(K key) {
```

Comment:
```suggestion
    private int findIndexByKey(K key) {
```
don't forget to sort methods by access modifier